File: Server/src/main/java/com/pbl3/musicapplication/model/model/RegisterRequest.java
package com.pbl3.musicapplication.model.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
public class RegisterRequest {
    private String username;
    private String password;
    private String fullName;
    private String phoneNumber;

    @Override
    public String toString() {
        return "RegisterRequest [username=" + username + ", password=" + password + ", fullName=" + fullName
                + ", phoneNumber=" + phoneNumber + "]";
    }
}

File: Vida/lib/models/register_request.dart
class RegisterRequest {
  String? username;
  String? password;
  String? fullName;
  String? phoneNumber;

  RegisterRequest(
      {this.username, this.password, this.fullName, this.phoneNumber});
  factory RegisterRequest.ByPassword(
      String userName, String password, String fullName, String phoneNumber) {
    return RegisterRequest(
        username: userName,
        password: password,
        fullName: fullName,
        phoneNumber: phoneNumber);
  }

  factory RegisterRequest.fromJson(Map<String, dynamic> json) {
    return RegisterRequest.ByPassword(json["username"], json["password"],
        json["fullName"], json["phoneNumber"]);
  }

  Map<String, dynamic> toJson() {
    return {
      "username": this.username,
      "password": this.password,
      "fullName": this.fullName,
      "phoneNumber": this.phoneNumber
    };
  }
}
